extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java

```diff
@@ -172,7 +174,10 @@ public void run() {
             closeBuildItem.addCloseTask(closeTask, true);
         }
         databases = runningDevServices;
-        cachedProperties = propertiesMap;
```

Comment:
I think that we could even remove the _propertiesMap_ entirely, and juste keep devDebProperties

Reply:
While I agree your solution looks cleaner, we need to think about which properties need to trigger a restart.

Previously, only a change in properties populated in `propertiesMap` would trigger a restart. Things like db kind, port, username, dev service properties, etc. Things that actually affect the dev service.

After your changes, any property starting with `quarkus.datasource` will trigger a restart of the container... including things like `quarkus.datasource.metrics.enabled` or `quarkus.datasource.jdbc.driver` or `quarkus.datasource.jdbc.min-size` ... which are clearly client-side?!


If you want to do things this way, I think you need to filter the maps obtained through `getOptionalValues` to only keep the properties that would actually matter for dev services (i.e. the ones we used to put in `propertiesMap`).